vm/qemu/qemu.go

```diff
@@ -220,6 +224,56 @@ var linuxCmdline = []string{
 	"biosdevname=0",
 }
 
+func (inst *instance) qmp_recv(conn net.Conn) (map[string]json.RawMessage, error) {
+	var qmp map[string]json.RawMessage
+
+	dec := json.NewDecoder(conn)
+	err := dec.Decode(&qmp)
+
+	return qmp, err
+}
+
+func (inst *instance) qmp(cmd string) (map[string]json.RawMessage, error) {
+	addr := fmt.Sprintf("127.0.0.1:%v", inst.monport)
```

Comment:
I think it's better to connect to the socket once and wrap it into json.Decoder once.
It can be nicely wrapped into a QmpConn object with qmp/hmp methods on it, currently these are methods on inst, but they don't use anything from inst.

Reply:
ok!